### PR TITLE
refactor: Dynamic datasource load (without datasource constants)

### PR DESCRIPTION
**Pull request by zharinov**
Ref #5419

### CODE REVIEW

File: lib/datasource/index.ts [hunk 1]
```diff
@@ -36,31 +14,37 @@ import { VERSION_SCHEME_SEMVER } from '../constants/version-schemes';
...
-  rubygems,
-  rubyVersion,
-  sbt,
-  terraform,
-  terraformProvider,
-};
+const datasources: Record<string, Datasource> = {};
+
+function isValidDatasourceModule(
+  datasourceName: string,
```
Comment (JamieMagee) [inline lib/datasource/index.ts]: unused?

File: lib/datasource/index.ts [hunk 2]
```diff
@@ -36,31 +14,37 @@ import { VERSION_SCHEME_SEMVER } from '../constants/version-schemes';
...
+    .filter(dirent => dirent.isDirectory())
+    .map(dirent => dirent.name)
+    .sort();
+  for (const datasourceName of datasourceDirs) {
+    let module = null;
+    try {
+      module = require(`./${datasourceName}`); // eslint-disable-line
+    } catch (err) /* istanbul ignore next */ {
+      logger.fatal({ err }, `Can not load datasource "${datasourceName}".`);
+      process.exit(1);
```
Comment (JamieMagee) [inline lib/datasource/index.ts]: #5421 ?

File: lib/datasource/index.ts [hunk 3]
```diff
@@ -36,31 +14,37 @@ import { VERSION_SCHEME_SEMVER } from '../constants/version-schemes';
...
+const datasources: Record<string, Datasource> = {};
+
+function isValidDatasourceModule(
+  datasourceName: string,
+  module: unknown
+): module is Datasource {
+  return !!module;
+}
+
+function loadDatasources(): void {
```
Comment (JamieMagee) [inline lib/datasource/index.ts]: Can you add test coverage for this method?

**Comment by rarkins** [comment]
Use #5440 instead

**Pull request closed without merging**